### PR TITLE
Delete studies legacy

### PR DESCRIPTION
**Pull request by anzioka**

### CODE REVIEW

**Review by solomontang — DISMISSED** [review]

**Review by bergieboy — DISMISSED** [review]

**Review by bergieboy — DISMISSED** [review]

**Review by solomontang — DISMISSED** [review]

File: src/containers/studies/StudiesSagas.js [hunk 1]
```diff
@@ -142,6 +145,50 @@ const LOG = new Logger('StudiesSagas');
...
+function* deleteStudyWorker(action :SequenceAction) :Saga<*> {
+  try {
+    yield put(deleteStudy.request(action.id));
+
+    const study = action.value;
+
+    const studyId = study.getIn([STUDY_ID, 0]);
+    const studyEKID = getEntityKeyId(study);
+
+    yield call(ChronicleApi.deleteStudy, studyId);
```
Comment (UnsungHero97) [inline src/containers/studies/StudiesSagas.js]: Why aren't you checking `response.error`?

File: src/containers/studies/StudiesSagas.js [hunk 2]
```diff
@@ -142,6 +145,50 @@ const LOG = new Logger('StudiesSagas');
 const CAFE_ORGANIZATION_ID = '7349c446-2acc-4d14-b2a9-a13be39cff93';
 const TIME_USE_DIARY = 'Time Use Diary';
 
+function* deleteStudyWorker(action :SequenceAction) :Saga<*> {
```
Comment (UnsungHero97) [inline src/containers/studies/StudiesSagas.js:145]: No change required. You should start putting sagas (and reducers) into their own files like we do with the orgs app.

File: src/containers/study/StudyDetails.js [hunk 3]
```diff
@@ -145,19 +151,37 @@ type Props = {
...
   const studyEmail = study.getIn([STUDY_EMAIL, 0]);
   const studyGroup = study.getIn([STUDY_GROUP, 0]);
 
+  useEffect(() => {
+    if (deleteStudyRS === RequestStates.SUCCESS) {
+      setTimeout(() => {
+        dispatch(removeStudyOnDelete(studyUUID));
+        dispatch(goToRoute(Routes.ROOT));
+        dispatch(resetRequestState(DELETE_STUDY));
+      }, 2000);
```
Comment (UnsungHero97) [inline src/containers/study/StudyDetails.js:170]: Why did you decide to go with `setTimeout` instead of handling this logic in the reducer?

**Review by UnsungHero97 — CHANGES_REQUESTED** [review]

File: src/containers/studies/StudiesSagas.js (hunk 1 above)
Comment (anzioka) [reply src/containers/studies/StudiesSagas.js]: No need to explicitly check for errors as they would still be propagated to the `catch` block

File: src/containers/study/StudyDetails.js (hunk 3 above)
Comment (anzioka) [reply src/containers/study/StudyDetails.js:170]: `setTimeout` gives the user an opportunity to see a successful deletion status message.

File: src/containers/studies/StudiesSagas.js (hunk 1 above)
Comment (UnsungHero97) [reply src/containers/studies/StudiesSagas.js]: Right! This is not `lattice-sagas`.

**Review by UnsungHero97 — DISMISSED** [review]

**Review by solomontang — DISMISSED** [review]

**Review by UnsungHero97 — APPROVED** [review]

**Review by solomontang — APPROVED** [review]

**Pull request merged**